client/js/shout.js

```diff
@@ -600,7 +613,9 @@ $(function() {
 		var button = sidebar.find(".chan[data-target=" + target + "]");
 		var isQuery = button.hasClass("query");
 		var type = msg.type;
-		var highlight = type.contains("highlight");
+		var highlight = type.contains("highlight")|| highlights.some(function(h) {
```

Comment:
So, having a look, the highlight type is included in the server message.js.

Obviously, I can't add this code there (because we don't have the highlights saved on the server just now, but that would be ideal for this when we get that done) but I've moved it slightly further up into the socket.on("msg") so that it's not all the way down here when the actual dom element changes. It's a bit nicer.